File: server/src/main/java/ru/trett/rss/models/Channel.java
package ru.trett.rss.models;

import jakarta.validation.constraints.NotEmpty;

import java.util.List;

public class Channel {

    public long id;

    /** RSS link */
    @NotEmpty public String channelLink;

    @NotEmpty public String title;

    /** Unique link for channel */
    @NotEmpty public String link;

    public List<Feed> feedItems = List.of();
}

File: server2/src/main/scala/ru/trett/server/models/Channel.scala
package ru.trett.server.models

import doobie.util.Read
import doobie.postgres.implicits._

case class Channel(
  id: Long,
  title: String,
  link: String,
  feedItems: List[Feed] = List.empty
)

object Channel {
  implicit val read: Read[Channel] = Read[(Long, String, String)].map {
    case (id, title, link) => Channel(id, title, link)
  }
}
